superset-frontend/src/SqlLab/components/ResultSet.tsx

```diff
@@ -87,6 +120,17 @@ export default class ResultSet extends React.PureComponent<
       searchText: '',
       showExploreResultsButton: false,
       data: [],
+      showSaveDatasetModal: false,
+      newSaveDatasetName: `${props.query.tab} ${moment().format(
+        'MM/DD/YYYY HH:mm:ss',
+      )}`,
```

Comment:
One thought that occurred me: if the query has a name should we use it to name the dataset?

Reply:
the tab = `Query {datasetName}` if the user is coming from explore view via `View in SqlLab`, We can look into just making it the dataset name